src/compiler/crystal/macros.cr

```diff
@@ -67,6 +67,19 @@ module Crystal::Macros
   def raise(message) : NoReturn
   end
 
+  # Reads a file: if it exists, returns a `StringLiteral` with its contents;
+  # otherwise `nil` is returned.
+  #
+  # To read a file relative to where the macro is defined, use:
+  #
+  # ```
+  # read_file("#{__DIR__}/some_file.txt")
```

Comment:
I think `./some_file.txt` should work, can you try it out? The idea is that this works like require.

Reply:
Hm, it probably works. But I don't think it should. Relative paths should be relative to the current working directory.
Either way, there should be a spec for reading a relative path.

Reply:
Well, we don't do require `"#{__DIR__}/..."`, we use relative paths. `require` can be thought as a macro call that requires the given file. `read_file` is another macro that works the same way but reads it at compile-time. Because it's compile-time, I think it makes sense to use the same rules as require. In fact, the whole change that @Sija made because of me requesting it is exactly that.

Reply:
The argument to `require` is always an (explicit) string literal. There is no reason to make it customizeable. And there is no reason to require a file relative to the working directory.

But to `read_file` you might want to pass a `StringLiteral` as argument that is originally defined in a completely different place. And it there are perfectly fine reasons to read a file relative to the current working directory instead of the source location.